src/score.rs

```diff
@@ -0,0 +1,37 @@
+use crate::{
+    frame::{Drawable, Frame},
+    invaders::Invaders,
+};
+
+pub struct Score {
+    chars: Vec<char>,
+}
+
+impl Score {
+    pub fn new() -> Self {
+        Self { chars: Vec::new() }
+    }
+
+    pub fn update(&mut self, invaders: &Invaders) {
+        // format our score string
+        let formated = format!("SCORE: {:0>3}", invaders.total_count - invaders.army.len());
```

Comment:
Very clever! This definitely _works_, but at the expense of tightly coupling `Score` to the internal workings of `Invaders`. 

What if kept them decoupled by reversing the approach -- instead of passing an `&Invaders` to `Score`'s `update(...)` what if we:
- Changed `Score` to store a numerical score
  - No `update()` method
  - With a `add_points(amount: u16)` method (or something like that)
  - Implement drawable by doing the string formatting of the stored numerical score and immediately writing that into the frame.
- Pass a `&mut Score` into `Invader`'s `update()` method.

Then `Invaders` remains the only thing that knows exactly what points have been generated during a frame, and it stores it in `Score`. `Score` doesn't even know about `Invaders`, it just knows how to keep track of the points and render them into the `Frame`.

Reply:
Yeah sounds good to me.

The only caveat I see is that the score will always be drawn which is fine for a simple game.
but it should only be "rendered" when the underlying score value changes. 

As a neat feature with the architecture we could give an `Invader` a `point` value which would allow the `Player.detect_hits` function to return an integer based on the hit `Invader`s worth and add it to the `Score`.

Maybe as another step in the tutorial. ;)